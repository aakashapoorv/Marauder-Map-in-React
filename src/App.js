import React from "react";
import FootSteps from "./FootSteps";
import MapFlap from "./MapFlap";
import MapSide from "./MapSide";

import "./styles.css";

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false
    };
  }
  toggleMap = () => {
    const currentState = this.state.active;
    this.setState({ active: !currentState });
  };
  render() {
    return (
      <div className="App">
        <div className="main-content" onClick={this.toggleMap}>
          <div className={this.state.active ? "map-base active" : "map-base"}>
            <FootSteps name="Severus Snape" number={1} />
            <FootSteps name="Harry Potter" number={2} />
            <MapFlap number={1} />
            <MapFlap number={2} />
            <MapSide side={1} map={8} isback={true} />
            <MapSide side={2} map={18} isback={true} />
            <MapSide side={3} map={7} isback={true} />
            <MapSide side={4} map={10} isback={false} />
            <MapSide side={4} map={10} isback={false} />
            <MapSide side={5} map={6} isback={true} />
            <MapSide side={6} map={11} isback={true} />
          </div>
        </div>
      </div>
    );
  }
}
