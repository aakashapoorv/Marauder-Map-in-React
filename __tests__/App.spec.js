import React from "react";
import App from "../src/App";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;

describe("App tests", () => {
  let AppElement;

  beforeEach(() => {
    AppElement = renderer.create(<App />);
  });

  it("Should Render correctly", () => {
    let tree = AppElement.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
