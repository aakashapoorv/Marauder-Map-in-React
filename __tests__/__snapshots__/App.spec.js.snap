// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Should Render correctly 1`] = `
<div
  className="App"
>
  <div
    className="main-content"
    onClick={[Function]}
  >
    <div
      className="map-base"
    >
      <div
        className="footsteps footsteps-1"
      >
        <div
          className="footstep left"
        />
        <div
          className="footstep right"
        />
        <div
          className="scroll-name"
        >
          <p>
            Severus Snape
          </p>
        </div>
      </div>
      <div
        className="footsteps footsteps-2"
      >
        <div
          className="footstep left"
        />
        <div
          className="footstep right"
        />
        <div
          className="scroll-name"
        >
          <p>
            Harry Potter
          </p>
        </div>
      </div>
      <div
        className="map-flap flap---1"
      >
        <div
          className="map-flap__front"
        />
        <div
          className="map-flap__back"
        />
      </div>
      <div
        className="map-flap flap---2"
      >
        <div
          className="map-flap__front"
        />
        <div
          className="map-flap__back"
        />
      </div>
      <div
        className="map-side side-1"
      >
        <div
          className="front map8"
        />
        <div
          className="back"
        />
      </div>
      <div
        className="map-side side-2"
      >
        <div
          className="front map18"
        />
        <div
          className="back"
        />
      </div>
      <div
        className="map-side side-3"
      >
        <div
          className="front map7"
        />
        <div
          className="back"
        />
      </div>
      <div
        className="map-side side-4"
      >
        <div
          className="front map10"
        />
      </div>
      <div
        className="map-side side-4"
      >
        <div
          className="front map10"
        />
      </div>
      <div
        className="map-side side-5"
      >
        <div
          className="front map6"
        />
        <div
          className="back"
        />
      </div>
      <div
        className="map-side side-6"
      >
        <div
          className="front map11"
        />
        <div
          className="back"
        />
      </div>
    </div>
  </div>
</div>
`;
